DKPY2 fixes to getting started, running examples, and developer setup…

hamishwillee opened the pull request:
This basically is first draft of changes to getting started, along with all the other minor link fixes an fixes to docs that made sense following the changes and which fix any bugs that showed up in links and rendering during the build. 

The getting started section is a lot simpler and I removed a lot of dependencies. Worth checking. I also removed the specific "running an example" section and instead now point to the examples section for that.

This does need a basic review.

tcr3dr commented:
Rebased to master.

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -21,7 +20,7 @@ Setting up the vehicle/autopilot
 For information on how to set up a vehicle (real and simulated) see:
 
 * :ref:`supported-companion-computers` for links to tested hardware/software configurations for a number of onboard Linux computers. 
-* :ref:`sitle_setup` for links explaining how to set up a simulated vehicle for Copter, Plane, or Rover.
```
`sitle`++

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -40,11 +39,10 @@ If you are using Ubuntu or Debian Linux you can get most of the *DroneKit* depen
 .. code:: bash
 
-    sudo apt-get install python-pip python-dev python-opencv python-serial python-pyparsing python-wxgtk2.8
+    sudo apt-get install python-pip python-dev
```
isn't this great?

tcr3dr replied:
> Pray to the God it fall down
> Pray to the cash it fall down

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
-.. code:: bash
+Set up a command line DroneKit-Python installation using *WinPython* (this Python distribution already includes most of the needed dependencies).
 
+<<<<<<< HEAD
```
I think you left this here from a git conflict, can you remove them please?

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
+<<<<<<< HEAD
     sudo pip install pyparsing
     sudo pip install dronekit
+=======
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
+   * Run the installer as an administrator (**Right-click** on file, select **Run as Administrator**). 
+   * When prompted for the destination location, specify **C:\\Program Files (x86)** 
+     (the default location is under the **Downloads** folder).
+>>>>>>> DKPY2 fixes to getting started, running examples, and developer setup windows. Consequent fixups to links elsewhere
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
-.. code:: bash
+.. tip::
 
+<<<<<<< HEAD
```
git conflict

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
 
+<<<<<<< HEAD
     echo "module load dronekit.module.api" >> ~/.mavinit.scr
+=======
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
+=======
+    The strings above are the same as you would use if connecting with MAVProxy. For other options see the 
+    `MAVProxy documentation <http://dronecode.github.io/MAVProxy/html/getting_started/starting.html#master>`_.
+>>>>>>> DKPY2 fixes to getting started, running examples, and developer setup windows. Consequent fixups to links elsewhere
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
-.. code:: bash
+.. code-block:: bash
 
+<<<<<<< HEAD
```
git conflict

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
 
+<<<<<<< HEAD
     echo module load dronekit.module.api >> %HOMEPATH%\AppData\Local\MAVProxy\mavinit.scr
+=======
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
     echo module load dronekit.module.api >> %HOMEPATH%\AppData\Local\MAVProxy\mavinit.scr
+=======
+    python your_dronekit_script.py
+>>>>>>> DKPY2 fixes to getting started, running examples, and developer setup windows. Consequent fixups to links elsewhere
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
 
-Alternatively you can choose to manually (re)load *DroneKit* into *MAVProxy* every time you need it:
 
+<<<<<<< HEAD
```
moaaar conflicts!

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
     MANUAL> module load dronekit.module.api
     DroneAPI loaded
     MANUAL>
+=======
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -61,151 +59,129 @@ installed when you get DroneKit-Python from the public PyPi repository:
+=======
+.. todo:: Connect method here needs to link to the function, but it isn't exported yet. Fix that once the API tidied.
+    
+>>>>>>> DKPY2 fixes to getting started, running examples, and developer setup windows. Consequent fixups to links elsewhere
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -214,6 +190,7 @@ Alternatively you can choose to manually (re)load *DroneKit* into *MAVProxy* eve
 Running an app/example
 ======================
 
+<<<<<<< HEAD
```
I'm guessing this isn't fun right?

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -289,5 +266,9 @@ The steps are:
        Got MAVLink msg: COMMAND_ACK {command : 11, result : 0}
        ...
 
+=======
```
same

mrpollo commented on docs/guide/getting_started.rst:
```diff
@@ -289,5 +266,9 @@ The steps are:
+=======
+This SDK has :ref:`numerous examples <example-toc>`. We recommend you start with :ref:`example-vehicle-state`, 
+which reads and writes :ref:`vehicle state and parameter <vehicle-information>` information. 
+>>>>>>> DKPY2 fixes to getting started, running examples, and developer setup windows. Consequent fixups to links elsewhere
```
same

mrpollo commented:
looks good but has some git conflicts that need to be resolved, possibly squash this commit afterwards too

tcr3dr commented:
Yeah the conflicts were probably me :) the only diff should be <<<OLD STUFF WITH "droneapi" replaced with "dronekit" ===== HAMISH'S CHANGES >>>>

mrpollo commented:
to be fair sphinx makes it really hard to understand what's going on

hamishwillee commented:
No guilty (me or sphinx). I'll try fix this up on Monday.

hamishwillee commented:
@tcr3dr @mrpollo I removed that rubbish. Admitting guilt <slap on wrist> :-(  I made sure that this also refers to dronekit rather than droneapi, and removed the numpy dependency.

Ready to merge (though I will hopefully add further updates to examples, depending on how testing of other fixes goes).
